submissions/antonkottans/friends-app/script.js

```diff
@@ -0,0 +1,595 @@
+    };
+
+    containerForFriends.innerHTML = "";
+    const fragment = createFragmentFromFriends();
+    containerForFriends.appendChild(fragment);
+    if (document.querySelector(".radio-nav-container"))
+        document
+            .querySelector(".radio-nav-container")
+            .replaceWith(createPagesNavigation(totalPages));
+    else {
+        document
+            .querySelector(".main")
+            .appendChild(createPagesNavigation(totalPages));
+    }
+
+    setTimeout(() => {
+        containerForFriends.childNodes.forEach((friendToShow) =>
+            friendToShow.classList.remove("hidden")
+        );
+    }, 0);
+};
+
+const doRecursiveFetch = (url, attempts) => {
+    if (attempts === 0) return;
+    return fetch(url)
+        .then((response) => {
+            if (response.ok) return response.json();
+        })
+        .catch((_) => {
+            return doRecursiveFetch(url, attempts - 1);
+        });
+};
+
+const downloadFriends = ({
+    pages = 5,
+    seed = "google",
+    friendsPerPage = 10,
+}) => {
+    const friendFieldsList = [
+        "name",
+        "dob",
+        "email",
+        "cell",
+        "id",
+        "gender",
+        "picture",
+    ];
+    let loadingProgressDots = [];
+
+    const placeLoadingIndicator = () => {
+        const loadingContainer = document.createElement("div");
+        loadingContainer.classList.add("loading-container");
+        const loadingText = document.createElement("span");
+        loadingText.classList.add("loading-text");
+        loadingText.insertAdjacentText("afterbegin", "Loading");
+        loadingProgressDots = [...new Array(pages)].map((_, i) => {
+            const dot = document.createElement("div");
+            dot.classList.add("in-progress", "loading-dot");
+            return dot;
+        });
+        loadingContainer.append(loadingText, ...loadingProgressDots);
+        containerForFriends.appendChild(loadingContainer);
+    };
+
+    placeLoadingIndicator();
+    state.loading = true;
+    friendsStorage.clearFriends();
+
+    Promise.all(
+        [...new Array(pages)].map((_, i) => {
+            return doRecursiveFetch(
+                `https://randomuser.me/api/?inc=${friendFieldsList}&seed=${seed}&page=${
+                    i + 1
+                }&results=${friendsPerPage}`
+            ).then((response) => {
+                loadingProgressDots[i].classList.remove("in-progress");
+                friendsStorage.addFriends(response.results);
+            });
+        })
+    ).then(() => {
+        state.loading = false;
+        updateContentAccordingToActiveFilters();
+    });
+};
+
+let createFriendCard = (friend) => {
+    const createFriendContainer = () => {
+        const friendCard = document.createElement("div");
+        friendCard.classList.add("friend-container", "hidden");
+
+        return friendCard;
+    };
+
+    const addName = (friendCard) => {
+        const friendName = document.createElement("span");
```

Comment:
If you will use `article` tag you should use the heading (h2 for example) for card title.